File: app/src/main/java/com/volkanhotur/basemvvm/android/base/exception/UnCaughtExceptionHandler.java
package com.volkanhotur.basemvvm.android.base.exception;

import android.app.Activity;

public class UnCaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
    private Activity activity;

    public UnCaughtExceptionHandler(Activity activity) {
        this.activity = activity;
    }
    @Override
    public void uncaughtException(Thread thread, Throwable ex) {
        try {
            Thread.sleep(50);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        activity.finishAffinity();
        android.os.Process.killProcess(android.os.Process.myPid());
        System.exit(1);
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/android/base/exception/UnCaughtExceptionHandler.kt
package com.volkanhotur.basemvvm.android.base.exception

import android.app.Activity
import android.os.Process
import kotlin.system.exitProcess

/**
 * @author volkanhotur
 */
class UnCaughtExceptionHandler(private val activity: Activity) : Thread.UncaughtExceptionHandler {
    override fun uncaughtException(thread: Thread, ex: Throwable) {
        try {
            Thread.sleep(50)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        activity.finishAffinity()
        Process.killProcess(Process.myPid())
        exitProcess(1)
    }

}